src/compiler/core.ts

```diff
@@ -2250,18 +2250,40 @@ namespace ts {
         }
     }
 
-    export function padLeft(s: string, length: number) {
-        while (s.length < length) {
-            s = " " + s;
+
+    /**
+     * Returns 's' left-padded with copies of 'padString' until it reaches the given 'length'.
+     * If omitted, 'padString' defaults to a single space.
+     * If 'length' is less than or equal to 's.length' or 'padString' is empty, returns 's' unchanged.
+     */
+    export function padLeft(s: string, length: number, padString = " ") {
+        if (length <= s.length || padString.length === 0) {
+            return s;
+        }
+        else {
+            length -= s.length;                 // how much padding we need
+            if (length > padString.length) {    // need more than one copy of padString
+                padString += padString.repeat(length / padString.length);
+            }
+            return padString.slice(0, length) + s;
         }
-        return s;
     }
 
-    export function padRight(s: string, length: number) {
-        while (s.length < length) {
-            s = s + " ";
+    /**
+     * Returns 's' right-padded with copies of 'padString' until it reaches the given 'length'.
+     * If omitted, 'padString' defaults to a single space.
+     * If 'length' is less than or equal to 's.length' or 'padString' is empty, returns 's' unchanged.
+     */
+    export function padRight(s: string, length: number, padString = " ") {

```

Comment:
Thanks for keeping the helpers in sync.

Reply:
In light of your previous comment about limiting this to `" "` or `"0"`, does it make sense to do this for `padRight()` anymore? Can't think of a use case where someone would want to pad a string with zeroes on the right... and yet, it _feels_ weird not to have the two functions match.

Reply:
Personally, I would have them match, but just change `" " | "0"` to `" "` in `padRight`.